ironfish/src/network/peerNetwork.ts

```diff
@@ -348,29 +344,42 @@ export class PeerNetwork {
     }
   }
 
-  private broadcastTransaction(
-    gossipMessage: NewTransactionMessage,
-    peerIdentity: string,
-  ): void {
+  private getPeersToSendTo(peerIdentity: string, hash: TransactionHash): ReadonlyArray<Peer> {
```

Comment:
This is refactoring out which peers to send the gossip message to. The only logic change is filtering out peers that have already seen the transaction hash. I put this into a separate function so that we can check if there are any peers that need the transaction before actually doing the intensive work of verifying and validating the transaction